### DISCUSSION TITLE
Compiler error: Stack too deep when compiling inline assembly: Variable value0 is 2 slot(s) too deep inside the stack.

### DISCUSSION THREAD

**Discussion by Jay-Sojitra** [crytic/slither#2268] [Q&A]
I want to find bugs in my contract so if I try on one contract named maxi then it is giving error like this =>
PS D:\Ethereum\Zeru_Internship\core-protocol> slither .\contracts\defiStrategies\strategies\maxi\Maxi.sol
'solc --version' running
'solc .\contracts\defiStrategies\strategies\maxi\Maxi.sol --combined-json abi,ast,bin,bin-runtime,srcmap,srcmap-runtime,userdoc,devdoc,hashes,compact-format --allow-paths .,D:\Ethereum\Zeru_Internship\core-protocol\contracts\defiStrategies\strategies\maxi' running
Compilation warnings/errors on .\contracts\defiStrategies\strategies\maxi\Maxi.sol:
Compiler error: Stack too deep when compiling inline assembly: Variable value0 is 2 slot(s) too deep inside the stack.

Traceback (most recent call last):
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\platform\solc.py", line 582, in _run_solc
    ret: Dict = json.loads(stdout)
                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Scripts\slither.exe\__main__.py", line 7, in <module>
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither\__main__.py", line 727, in main
    main_impl(all_detector_classes=detectors, all_printer_classes=printers)
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither\__main__.py", line 833, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither\__main__.py", line 96, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 718, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 207, in __init__
    self._compile(**kwargs)
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 629, in _compile
    self._platform.compile(self, **kwargs)
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\platform\solc.py", line 151, in compile
    targets_json = _get_targets_json(compilation_unit, self._target, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\platform\solc.py", line 280, in _get_targets_json
    return _run_solc(
           ^^^^^^^^^^
  File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\platform\solc.py", line 586, in _run_solc
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Compiler error: Stack too deep when compiling inline assembly: Variable value0 is 2 slot(s) too deep inside the stack.

PS D:\Ethereum\Zeru_Internship\core-protocol>

=> and also I have checked with npx hardhat compile then all contracts are compiling without any error. 
so what is the solution of this problem?

**Comment by elopez** [comment]
Hi! Use `slither .` (i.e. pass the hardhat project folder and not just a `.sol` file) to use hardhat to compile your contracts.

If you want to use a standalone contract file, you may need to enable solc compiler optimizations or refactor your code so that it builds, right now it does not:

```
Compilation warnings/errors on .\contracts\defiStrategies\strategies\maxi\Maxi.sol:
Compiler error: Stack too deep when compiling inline assembly: Variable value0 is 2 slot(s) too deep inside the stack.
```

**Reply by Jay-Sojitra** [reply]
but slither . is also not working it is giving me this error =>
PS D:\Ethereum\Zeru_Internship\core-protocol> slither .
'npx buidler compile' running

An unexpected error occurred:
helpers/misc-utils.ts(137,16): error TS2769: No overload matches this call.
Overload 1 of 2, '(o: {}): string[]', gave the following error.
Argument of type 'T' is not assignable to parameter of type '{}'.
Overload 2 of 2, '(o: object): string[]', gave the following error.
Argument of type 'T' is not assignable to parameter of type 'object'.

Traceback (most recent call last):
File "", line 198, in run_module_as_main
File "", line 88, in run_code
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Scripts\slither.exe_main.py", line 7, in
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither_main.py", line 727, in main
main_impl(all_detector_classes=detectors, all_printer_classes=printers)
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither_main_.py", line 833, in main_impl
) = process_all(filename, args, detector_classes, printer_classes)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\slither_main_.py", line 96, in process_all
compilations = compile_all(target, **vars(args))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 718, in compile_all
compilations.append(CryticCompile(target, **kwargs))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 207, in init
self._compile(**kwargs)
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\crytic_compile.py", line 629, in _compile
self._platform.compile(self, **kwargs)
File "C:\Users\Jay\AppData\Local\Programs\Python\Python311\Lib\site-packages\crytic_compile\platform\buidler.py", line 97, in compile
raise InvalidCompilation(txt)
crytic_compile.platform.exceptions.InvalidCompilation: buidler compile failed. Can you run it?
.\cache\solc-output.json not found
PS D:\Ethereum\Zeru_Internship\core-protocol>
and my hardhat.config.ts file is here =>

import path from 'path';
import fs from 'fs';
import { HardhatUserConfig } from 'hardhat/types';
// @ts-ignore
import { accounts } from './test-wallets.js';
import {
  eAvalancheNetwork,
  eEthereumNetwork,
  eNetwork,
  ePolygonNetwork,
  eBinanceNetwork,
  eXDaiNetwork,
  eArbitrumNetwork,
} from './helpers/types';
import { BUIDLEREVM_CHAINID, COVERAGE_CHAINID } from './helpers/buidler-constants';
import {
  NETWORKS_RPC_URL,
  NETWORKS_DEFAULT_GAS,
  BLOCK_TO_FORK,
  buildForkConfig,
} from './helper-hardhat-config';

require('dotenv').config();

import '@nomiclabs/hardhat-ethers';
import '@nomiclabs/hardhat-waffle';
// import 'temp-hardhat-etherscan';
import 'hardhat-gas-reporter';
import 'hardhat-typechain';
import 'solidity-coverage';
import { fork } from 'child_process';
import '@nomicfoundation/hardhat-verify';

import * as tdly from '@tenderly/hardhat-tenderly';

const SKIP_LOAD = process.env.SKIP_LOAD === 'true';
const DEFAULT_BLOCK_GAS_LIMIT = 9000000000;
const DEFAULT_GAS_MUL = 5;
const HARDFORK = 'istanbul';
const ETHERSCAN_KEY = process.env.ETHERSCAN_KEY || '';
const MNEMONIC_PATH = "m/44'/60'/0'/0";
const MNEMONIC = process.env.MNEMONIC || '';
const UNLIMITED_BYTECODE_SIZE = process.env.UNLIMITED_BYTECODE_SIZE === 'true';

// Prevent to load scripts before compilation and typechain
if (!SKIP_LOAD) {
  console.log('Inside of SKIP_LOAD'),
    [
      'misc',
      'migrations',
      'dev',
      'full',
      'verifications',
      'deployments',
      'helpers',
      'reservegrowth',
      'test',
      'strategies',
    ].forEach((folder) => {
      const tasksPath = path.join(__dirname, 'tasks', folder);
      fs.readdirSync(tasksPath)
        .filter((pth) => pth.includes('.ts'))
        .forEach((task) => {
          require(`${tasksPath}/${task}`);
        });
    });
}

require(`${path.join(__dirname, 'tasks/misc')}/set-bre.ts`);

const getCommonNetworkConfig = (networkName: eNetwork, networkId: number) => ({
  url: NETWORKS_RPC_URL[networkName],
  hardfork: HARDFORK,
  blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
  gasMultiplier: DEFAULT_GAS_MUL,
  gasPrice: NETWORKS_DEFAULT_GAS[networkName],
  chainId: networkId,
  accounts: {
    mnemonic: MNEMONIC,
    path: MNEMONIC_PATH,
    initialIndex: 0,
    count: 20,
  },
});

let forkMode;

const buidlerConfig: HardhatUserConfig = {
  solidity: {
    version: '0.8.0',
    settings: {
      optimizer: { enabled: true, runs: 200 },
      evmVersion: 'istanbul',
    },
  },
  typechain: {
    outDir: 'types',
    target: 'ethers-v5',
  },
  etherscan: {
    apiKey: ETHERSCAN_KEY,
  },
  mocha: {
    timeout: 0,
  },
  tenderly: {
    project: process.env.TENDERLY_PROJECT || '',
    username: process.env.TENDERLY_USERNAME || '',
    forkNetwork: '1', //Network id of the network we want to fork
  },
  networks: {
    coverage: {
      url: 'http://127.0.0.1:8545',
      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
      gas: 100000000,
      gasPrice: 20000000000,
      allowUnlimitedContractSize: UNLIMITED_BYTECODE_SIZE,
      chainId: BUIDLEREVM_CHAINID,
      timeout: 400000000,
      forking: buildForkConfig(),
      hardfork: 'london',
    },
    localhost: {
      url: 'http://127.0.0.1:8545',
      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
      gas: 10000000,
      gasPrice: 20000000000,
      allowUnlimitedContractSize: UNLIMITED_BYTECODE_SIZE,
      chainId: 31337,
      timeout: 400000000,
      forking: buildForkConfig(),
    },

    kovan: getCommonNetworkConfig(eEthereumNetwork.kovan, 42),
    ropsten: getCommonNetworkConfig(eEthereumNetwork.ropsten, 3),
    main: getCommonNetworkConfig(eEthereumNetwork.main, 1),
    tenderly: getCommonNetworkConfig(eEthereumNetwork.tenderly, 3030),
    testbsc: getCommonNetworkConfig(eBinanceNetwork.testbsc, 97),
    matic: getCommonNetworkConfig(ePolygonNetwork.matic, 137),
    mumbai: getCommonNetworkConfig(ePolygonNetwork.mumbai, 80001),
    xdai: getCommonNetworkConfig(eXDaiNetwork.xdai, 100),
    avalanche: getCommonNetworkConfig(eAvalancheNetwork.avalanche, 43114),
    fuji: getCommonNetworkConfig(eAvalancheNetwork.fuji, 43113),
    arbitrumGoerli: getCommonNetworkConfig(eArbitrumNetwork.goerli, 421613),
    hardhat: {
      hardfork: 'berlin',
      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
      gas: DEFAULT_BLOCK_GAS_LIMIT,
      gasPrice: DEFAULT_BLOCK_GAS_LIMIT,
      allowUnlimitedContractSize: UNLIMITED_BYTECODE_SIZE,
      chainId: BUIDLEREVM_CHAINID,
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      accounts: accounts.map(({ secretKey, balance }: { secretKey: string; balance: string }) => ({
        privateKey: secretKey,
        balance,
      })),
      forking: buildForkConfig(),
    },
    buidlerevm_docker: {
      hardfork: 'berlin',
      blockGasLimit: 9500000,
      gas: 9500000,
      gasPrice: 8000000000,
      chainId: BUIDLEREVM_CHAINID,
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      url: 'http://localhost:8545',
    },
    ganache: {
      url: 'http://ganache:8545',
      accounts: {
        mnemonic: 'fox sight canyon orphan hotel grow hedgehog build bless august weather swarm',
        path: "m/44'/60'/0'/0",
        initialIndex: 0,
        count: 20,
      },
    },
  },
};

export default buidlerConfig;

=> should i change something?


![image](https://github.com/crytic/slither/assets/84234572/6ee6ed09-3b06-4682-93d2-a0110db2683d)
